## 1. Import the Pandas Library and Read the Dataset File in Spreadsheet Format based on the First - Fifth Debate Videos

### 1.1. DataFrame Variable from First Debate Video on KOMPASTV YouTube Channel

In [1]:
"""
Melakukan proses import library Pandas serta memanggil file dataset dari hasil
web scraping komentar video YouTube debat capres - cawapres babak pertama di channel YouTube KOMPASTV
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3 yang telah
disimpan atau di-export pada file berformat spreadsheet yang bernama "Export Comments Data from 1st Video Debate.xlsx"
"""
import pandas as pd


first_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 1st Video Debate.xlsx")

first_debate_video_df.head(9) # Menampilkan 9 data tertas yang dimiliki oleh variabel DataFrame "first_debate_video_df"

video_id                channel_id                  comment_id  \
0  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzTOJaoDFAy69VEmW54AaABAg   
1  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzmTSpOSPp8Aaid8tN4AaABAg   
2  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgytfyaQRBKOJD1MWmJ4AaABAg   
3  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMdksVHA_TUd0b8zZ4AaABAg   
4  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzKN65yuzOuwoV25xR4AaABAg   
5  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwkGCjnkn4qiL2pUq54AaABAg   
6  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwHT6Z026fOcLqsxT54AaABAg   
7  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwEn2xOZrliv0jvyhp4AaABAg   
8  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwpSjBjJZnKzyn5PmN4AaABAg   

                 author                                       text_comment  \
0    @alkhasanisani3619  Anies pinter ngomong<br>Ganjar pinter jawab<br...   
1       @MrGaming-fg9qu  Indonesia tidak kekurangan orang orang pintar&...   
2    @desyanggraeni9380  Disaat Ganjar dan Anies saling menyerang Prabo...   
3     @aprianisalam7842  Pak Prabowo memang kurang dr hal seperti pak g...   
4     @nanananina-eg5vo  Jendral tidak dilatih untuk merangkai kata2 ma...   
5  @RidhoRamadhan-gm6rp  Saya tau rasanya menjadi orang seperti pak pro...   
6      @user-mf1tc7co8s  Pak prabowo tu keras , disiplin karena mantan ...   
7     @nanasepdiana5531  Pak prabowo tidak pandai merangkai kata - kata...   
8          @FauzanFikli  Saya yakin .pak Prabowo sudah tahu keadaan neg...   

   like_count  total_reply                                  channel_url  \
0         772          154    http://www.youtube.com/@alkhasanisani3619   
1         607           43       http://www.youtube.com/@MrGaming-fg9qu   
2         474           86    http://www.youtube.com/@desyanggraeni9380   
3         450           77     http://www.youtube.com/@aprianisalam7842   
4         431           56     http://www.youtube.com/@nanananina-eg5vo   
5         409           28  http://www.youtube.com/@RidhoRamadhan-gm6rp   
6         338           46      http://www.youtube.com/@user-mf1tc7co8s   
7         330           38     http://www.youtube.com/@nanasepdiana5531   
8         318           36          http://www.youtube.com/@FauzanFikli   

       published_date        updated_date  
0 2023-12-13 13:26:17 2023-12-13 13:26:17  
1 2023-12-23 15:19:21 2023-12-23 15:19:21  
2 2023-12-15 09:21:02 2023-12-15 09:21:36  
3 2023-12-14 12:31:52 2023-12-14 12:31:52  
4 2023-12-17 03:43:34 2023-12-17 03:43:34  
5 2023-12-27 14:30:09 2023-12-27 14:30:09  
6 2023-12-14 01:33:20 2023-12-14 01:33:20  
7 2023-12-26 08:02:26 2023-12-26 08:02:26  
8 2023-12-15 16:14:53 2023-12-15 16:14:53

In [2]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.shape

(10500, 10)

In [3]:
# Menampilkan 28 data terbawah yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.tail()

video_id                channel_id                  comment_id  \
10495  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugy824X3qgcCGYQKpJh4AaABAg   
10496  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgypIhXiDhi5Xr7cNP54AaABAg   
10497  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugzv4QLhWKkBwB-OoRF4AaABAg   
10498  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyS2Cqh17pCtZvDED54AaABAg   
10499  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxQv_F3aWzlQafGTzZ4AaABAg   

                      author  \
10495  @gitadesiamiranda2187   
10496       @user-le6ex4xf9m   
10497       @user-le6ex4xf9m   
10498      @athanalfariz3344   
10499      @malikibrahim8801   

                                            text_comment  like_count  \
10495  Paslon 1 dan 3 bercerita bukan menyampaikan VI...           0   
10496                Gk jadi milih 1, aku milih 4 ajah 😊           0   
10497       Pak hapuskan dunia pendidikan saya COBLOS 🔥🔥           0   
10498  Kata almarhum Gus Dur,,, Prabowo lah yang pali...           0   
10499  Sebenernya ini klwrga jkowi sdg di permalukan ...           0   

       total_reply                                   channel_url  \
10495            0  http://www.youtube.com/@gitadesiamiranda2187   
10496            0       http://www.youtube.com/@user-le6ex4xf9m   
10497            0       http://www.youtube.com/@user-le6ex4xf9m   
10498            0      http://www.youtube.com/@athanalfariz3344   
10499            0      http://www.youtube.com/@malikibrahim8801   

           published_date        updated_date  
10495 2023-12-23 01:46:20 2023-12-23 01:46:20  
10496 2023-12-23 01:46:24 2023-12-23 01:46:24  
10497 2023-12-23 01:47:07 2023-12-23 01:47:07  
10498 2023-12-23 01:49:39 2023-12-23 01:49:39  
10499 2023-12-18 16:31:23 2023-12-18 16:31:23

### 1.2. DataFrame Variable from Second Debate Video on KOMPASTV YouTube Channel

In [4]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kedua di channel YouTube KOMPASTV
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

second_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 2nd Video Debate.xlsx")

second_debate_video_df.head(15) # Menampilkan 15 data teratas yang dimiliki oleh variabel DataFrame "second_debate_video_df"

video_id                channel_id                  comment_id  \
0   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgzXUiEcKGGoUX0nDeh4AaABAg   
1   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugxjg4pvN724Yyi8jU14AaABAg   
2   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgzVOs7T9EaLxEPJjNN4AaABAg   
3   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgyBOQDTIbXXmH8m1Vx4AaABAg   
4   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugypp9rnOjLTrXH5Tb14AaABAg   
5   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugzjx9OYz33QH4J6IlF4AaABAg   
6   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgxnjEQN6PyIZkF1ZT54AaABAg   
7   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugz0rfvHOz9Zzk_rcqp4AaABAg   
8   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugz98U4599KOyDouo0h4AaABAg   
9   92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgyHMsJH7P7454VXaD54AaABAg   
10  92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgzXF0rdRzDFI7iEHEh4AaABAg   
11  92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugz8CbXZ6v7jTuZ1LUJ4AaABAg   
12  92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgxK7mC17_YxaD4tdmd4AaABAg   
13  92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  Ugzo2gl0Dw47lE1l-Zx4AaABAg   
14  92k51DVSDv4  UC5BMIWZe9isJXLZZWPWvBlg  UgwW8aTfOYP9AZ3eGWh4AaABAg   

                          author  \
0            @ariessusanti-ep2or   
1                    @lena_palma   
2                   @hidayat9478   
3                    @melisa8319   
4           @anisatulainiyah1001   
5              @wargaproxima7105   
6              @hermanussius6712   
7                 @ajiwijaya4779   
8          @PariyemTarisna-dg5pk   
9               @user-ic5ln3do2u   
10               @liayuliani1130   
11             @andivalencia7889   
12                @gitaprimsa970   
13  @sulhanudinelectronickit3024   
14                 @tiaraputri66   

                                         text_comment  like_count  \
0   Gibran ternyata gak spt yg disangka sangka, pe...         121   
1   Gibran SAVAGE🫶🏻🔥 cocok jadi simbol perjuangan ...         114   
2   Sudah saatnya kita butuh pemimpin yang fresh, ...         101   
3   Sempet skeptis sama gibran karna bantuan paman...          87   
4   Serius merinding dengar gibran jawab. Secerdas...          85   
5   Anak muda memang berfikirnya praktis, tidak mu...          76   
6   Gibran  mantap sebagai wapres,  Indonesia maju...          74   
7   Mahpud : bukan ngwbahas visi misi program kerj...          53   
8   Mantap mas Gibran ,maju terus ,semoga mas gibr...          49   
9   Mas Gibran anak muda yang tegas berani luar bi...          48   
10  Mas gibran seakan membuktikan bahwa dia pantas...          42   
11  kmrn masih gak yakin, tp akhirnya yakin pilih ...          38   
12  Sejujurnya dari beberapa topik pertanyaan itu ...          37   
13                saya yakin 100%  paslon no.2 menang          32   
14  Butuh bbrp menit biar bisa mencerna pertanyaan...          29   

    total_reply                                        channel_url  \
0            21         http://www.youtube.com/@ariessusanti-ep2or   
1             8                 http://www.youtube.com/@lena_palma   
2            18                http://www.youtube.com/@hidayat9478   
3             6                 http://www.youtube.com/@melisa8319   
4            18        http://www.youtube.com/@anisatulainiyah1001   
5             8           http://www.youtube.com/@wargaproxima7105   
6            10           http://www.youtube.com/@hermanussius6712   
7             7              http://www.youtube.com/@ajiwijaya4779   
8             3       http://www.youtube.com/@PariyemTarisna-dg5pk   
9             1            http://www.youtube.com/@user-ic5ln3do2u   
10            3             http://www.youtube.com/@liayuliani1130   
11            0           http://www.youtube.com/@andivalencia7889   
12            7              http://www.youtube.com/@gitaprimsa970   
13            6  http://www.youtube.com/@sulhanudinelectronicki...   
14            0               http://www.youtube.com/@tiaraputri66   

        published_date        updat

In [5]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "second_debate_video_df"
second_debate_video_df.shape

(600, 10)

### 1.3. DataFrame Variable from Third Debate Video on KOMPASTV YouTube Channel

In [6]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat
Microsoft Excel (spreadsheet) dari hasil web scraping komentar video YouTube
debat capres - cawapres babak ketiga di channel YouTube KOMPASTV
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

third_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 3rd Video Debate.xlsx")

third_debate_video_df.head(19)  # Menampilkan 19 data teratas yang dimiliki oleh variabel DataFrame "third_debate_video_df"

video_id                channel_id                  comment_id  \
0   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgzsQhZqQAKIiJ1V4kV4AaABAg   
1   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwJizMS3lAYMd08EMB4AaABAg   
2   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwkSPCre6psFjoGPaV4AaABAg   
3   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  Ugyc18GNff019r2AR2J4AaABAg   
4   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgyDE99FMiocfmsOwsh4AaABAg   
5   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  Ugy9RZWZmf8FMdw58TZ4AaABAg   
6   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgxD-1ZT_Zm0w8OT7F14AaABAg   
7   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  Ugyn8-ckOPuHc6gOipt4AaABAg   
8   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgyvDyhafDoNI268k9V4AaABAg   
9   FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwbgqYzFwqxqLFK7_N4AaABAg   
10  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  Ugzi1Zy612T-50Lr_7d4AaABAg   
11  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgyLcCk9vulEokZjnUt4AaABAg   
12  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  Ugyt9zG0zVYUSGFfJjd4AaABAg   
13  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwcFB0xGmAyqjUnVMp4AaABAg   
14  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgxarGy1lSFU8JR0_s54AaABAg   
15  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwBvTBMA9R0GSwLPA94AaABAg   
16  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgxCpDG1WpVyKGN_Ec94AaABAg   
17  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgwoPtVTmo9jK_igWu54AaABAg   
18  FKImOfLQTbw  UC5BMIWZe9isJXLZZWPWvBlg  UgyyhhON8AvwuDftDwB4AaABAg   

                    author                                       text_comment  \
0              @hanafi5541  Prabowo berdiri sendiri karena niat baiknya un...   
1       @setiadysastra5220  Bapak prabowo itu capres paling ikhlas untuk r...   
2           @katarinae4452  Jadi warga Jakarta bertahun- tahun sudah meras...   
3               @fitra8968  Sangat kelihatan sekali wajah Bapak Prabowo yg...   
4         @abdullahhhhh176  Demi menang di pilpres ini Bapak membahayakan ...   
5         @user-xb2fh5kq1o  Saya orang jawa tengah... anti GANJAR... makin...   
6         @a.r.pratama6708  Debat ini Prabowo bahas negara sedangkan Anis ...   
7      @BudiHartanto-xk3xi  Yg sbr pk prabowo ....<br>Kmi sbgai rakyat tw ...   
8                @YasintaL  Pada moment ini Paslon 1 dan 3, dengan semanga...   
9         @ariyosaloko2281  Prabowo telah diuntungkan oleh Tema Debat..sem...   
10  @yettidwiagustiani1637  Klau liat muka wan abut ini kesel bngt sumpah ...   
11        @marthenruru7600  Industri pertahanan jaman Prabowo sdh maju pes...   
12        @NurHayati-do5ze  Sehat sll pak probowo kami NTB Sangat mendukun...   
13          @nuratikah6523  Menteri yang di pecat memberi nilai kepada sla...   
14         @PojokChannel99  JANGAN PERNAH MENINGGALKAN KALIMATU &quot;BISM...   
15             @binti_0924  Pak Prabowo sudah memberikan bukti bekerja bai...   
16         @dheaasyifa3206  maasyaaAllah begitu selesai acara debat dan sa...   
17           @ayusyifa5443  Anis pinter omong ya 🤭<br>Dr awal sampe akhir ...   
18         @unikjanuar2358                 Saya malah salut sama pak Prabowo,   

    like_count  total_reply                                    channel_url  \
0           73           15             http://www.youtube.com/@hanafi5541   
1           71           14      http://www.youtube.com/@setiadysastra5220   
2           65           16          http://www.youtube.com/@katarinae4452   
3           60            9              http://www.youtube.com/@fitra8968   
4           60            4        http://www.youtube.com/@abdullahhhhh176   
5           57            3        http://www.youtube.com/@user-xb2fh5kq1o   
6           54           16        http://www.youtube.com/@a.r.pratama6708   
7           53            3     http://www.youtube.com/@BudiHartanto-xk3xi   
8           50           38               http://www.youtube.com/@YasintaL   
9           42           10        http://www.youtube.com/@ariyosaloko2281   
10          42            8  http://www.youtube.com/@yettidwiagusti

In [7]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "third_debate_video_df"
third_debate_video_df.shape

(600, 10)

### 1.4. ataFrame Variable from Fourth Debate Video on KOMPASTV YouTube Channel

In [8]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak keempat di channel YouTube KOMPASTV
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fourth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 4th Video Debate.xlsx")

fourth_debate_video_df.head(38) # Menampilkan 38 data teratas yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"

video_id                channel_id                  comment_id  \
0   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgybmbmGb4jA3yBpJKh4AaABAg   
1   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxcXhLRdRwawyNefKJ4AaABAg   
2   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxXBns_Vco13Tb9kRJ4AaABAg   
3   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugzv8oZwAHvc29avDYZ4AaABAg   
4   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgzYD0sGCrRAM6KkLpd4AaABAg   
5   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwKQnFYqsDUceLPiIt4AaABAg   
6   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxJ9Eye6mjc22dWCDB4AaABAg   
7   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwFAJvF9cqF6dpSLgF4AaABAg   
8   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxplxcxqKzDXq20FSZ4AaABAg   
9   _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyNtBuk2VCnmv72GKh4AaABAg   
10  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwWp0dwKMF4xrkKgwx4AaABAg   
11  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxaI011eHlw2gTB8mZ4AaABAg   
12  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwnJh_y8xxCPrWDwxp4AaABAg   
13  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgzwSZpotT26Bo3rvlR4AaABAg   
14  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxYuOlErft2sYpd1fx4AaABAg   
15  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugy1gKkyq_qlH8fcdV54AaABAg   
16  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwkDcwLxRghSNkKul54AaABAg   
17  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgymccLSD7ENJ0wKqHx4AaABAg   
18  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugz_XBxupzlKm7pX-xR4AaABAg   
19  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgzWM5WKQe-o0cG7AIh4AaABAg   
20  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugx-Dm5fC51t-Gvkx1N4AaABAg   
21  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugw_AmqjfmKsi_v-YiZ4AaABAg   
22  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxEH4K5-CWkK9nuytV4AaABAg   
23  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxqIwqXTzd6qr4cKD14AaABAg   
24  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyjFh23TjnHCzGPFRh4AaABAg   
25  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgwbfAhzIOVVweRYabV4AaABAg   
26  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugx5g21VwLL1gvoLS_J4AaABAg   
27  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyLnlMynExLqhZVnqx4AaABAg   
28  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugyog3c5csR3-kKgUDJ4AaABAg   
29  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugz6ou2ZkUfmJ5D4KBV4AaABAg   
30  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugzi7suoaMTVxbYI3P54AaABAg   
31  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugwv2qtSt_1uas_UnGl4AaABAg   
32  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgxPBhyft6PaOVAzgth4AaABAg   
33  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyHQ4l5f8tHbUVsrVZ4AaABAg   
34  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugzg0nZIimoCv9jW7aZ4AaABAg   
35  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyUnr6qdHK0nGCa9MZ4AaABAg   
36  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  UgyF-a7OEsOPddP6AYR4AaABAg   
37  _V033ckLwcE  UC5BMIWZe9isJXLZZWPWvBlg  Ugx8xBXH0idF8TmpnX94AaABAg   

                          author  \
0                @gibran_abidin1   
1             @ahmadnahareverend   
2       @riobinyobillionaire4523   
3                    @yonasboyke   
4               @RahmaWati-sr8ft   
5              @hidayat_abdullah   
6           @sarsukuakitriau7950   
7                 @gaduh_wacana4   
8                @fannyoctaviany   
9             @gangsar_sihombing   
10                    @djjll7026   
11                    @djjll7026   
12                @abdulroni1300   
13  @madukaliandramekarabadi1698   
14                    @djjll7026   
15                @javabrown4851   
16                @cahayalain208   
17                 @jimmycamario   
18                    @efrizal12   
19                    @djjll7026   
20                      @syeefah   
21                   @AnomSyurot   
22                     @erik_R.I   
23                 @turaya_maisa   
24                        @dsyeu   
25             @ahmadsyahrir2246   
26               @syahmariam2859   
27            @emiliasumbawa5533   
28                  @tvituna1920   
29                 @delvia_askia   
30                @widyawati3268   
31     

In [9]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"
fourth_debate_video_df.shape

(300, 10)

### 1.5. DataFrame Variable from Fifth Debate Video on KOMPASTV YouTube Channel

In [10]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kelima di channel YouTube KOMPASTV
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fifth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 5th Video Debate.xlsx")

fifth_debate_video_df.head(27) # Menampilkan 27 data teratas yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"

video_id                channel_id                  comment_id  \
0   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzYR03sxvD9SmDCx8t4AaABAg   
1   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwNO6z3M4ztyp65tZJ4AaABAg   
2   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxRc4CbpmeHg6ora4x4AaABAg   
3   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgyM8XhGIVg_v60IzHt4AaABAg   
4   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugw0wk0CfJ7vrcrkxuZ4AaABAg   
5   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxzCXZW6nOqz7U5Ti14AaABAg   
6   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugyl6Ep19otH8KL8NjZ4AaABAg   
7   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxfknZEYano0O6erud4AaABAg   
8   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzzdYFztVPj5ETA2FF4AaABAg   
9   uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxnBlOf3rx3920zNGl4AaABAg   
10  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugx-ATpfXpnMCYemWgF4AaABAg   
11  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzrCu0VFtFexJdJnuZ4AaABAg   
12  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgyVOJYdKNv32QJSIBl4AaABAg   
13  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgyJLpFd_jRFsWW2HGd4AaABAg   
14  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxjzJ9JepJho4MMUoZ4AaABAg   
15  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugxe1iUX7rv0wCQ3h1B4AaABAg   
16  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugw4pbDDzHUQnJAXGOZ4AaABAg   
17  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxfU2aQmb7kXFGrxUh4AaABAg   
18  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwU2NOR3bf9elpbXZJ4AaABAg   
19  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxZrzYp9pGo8HYYdWV4AaABAg   
20  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgyTZ-gm2UJkB1-tQZN4AaABAg   
21  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxzaZeWOdffKT7hiXN4AaABAg   
22  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxQNIOsKnWT0tp9M9N4AaABAg   
23  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzShsT-ITS7oW_EWtF4AaABAg   
24  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugy50tqwa8nxUdOzfrN4AaABAg   
25  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugw5MAtW75IME2Ux3yd4AaABAg   
26  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwgsYqlAshnps7acYF4AaABAg   

                 author                                       text_comment  \
0        @ikalpalem1811  Plato pernah bilang, jika seseorang kalah dala...   
1    @riskaandriani3066  Terlihatlah baik bukan karna menjelekan orang....   
2            @Garuda762  Semua paslon menginginkan yg terbaik untuk Ind...   
3     @muhamadidris2956  Kami rakyat indonesia butuh kenyataan dari teo...   
4            @feliciaMT  pada sesi tanya jawab, paslon 03 terlalu sibuk...   
5     @fitriyahammar648  Visi misi itu hanya pemanis mreka utk menarik ...   
6      @user-cv4cq6fv1r  Sebelum debat  terakhir pilihan saya pak Anis....   
7      @user-vb5wx1fm7d  pidato penutup pak Anis , masya Allah luar bia...   
8        @anafia.jalali  Kerennnn pak prabowo solutif dan action, jawab...   
9        @yansaizyn1681  Kami rakyat indonesi setuju dengan setuju deng...   
10         @EfrianiAris                     Amin untuk Indonesia perubahan   
11       @garryblind109  kurasa masyarakat sudah pada pintar untuk meni...   
12     @user-xc8dp5xh6x                             Salam 01 Dari Surabaya   
13     @arizalbarito649  Sangat Salut sama Pak Prabowo, tetap rendah ha...   
14    @dedekusmawan5961                                       Semakin amin   
15     @daengsiriwa7149  Pak Anis berusaha dengan kemampuan dan potensi...   
16     @user-ko7lw2of4s  Anis memang pintar bicara, itubaruhanyabicara,...   
17   @aldinaaprilia3485  Keren visi misi ny pak prabowo, sngt jelas mud...   
18              @izka25  Kok sy liat pak Prabowo lebih banyak aksi yah,...   
19          @Hicksss946  Luar biasa Pak Prabowo pernyataan penutupnya ,...   
20            @zuncidro  Aq terharu melihat 3 calon ini..  Semoga yg me...   
21        @evihasan1349            Semakin mantap memilih Anies Baswedan 😊   
22   @jackaksaputra4360  Teringat kata buya Arrazy mungkin hati pak Pra...   
23           @save-7964  01 Kelebihan Gula (Halu)<br>02 Netral (Tulus)<...   
24  @johanal

In [11]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.shape

(500, 10)

In [12]:
# Menampilkan 11 data terbawah yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.tail(11)

video_id                channel_id                  comment_id  \
489  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugxb0kUlJ-q5lKD5Urd4AaABAg   
490  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxTJgAR-MZOsCb_GvR4AaABAg   
491  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzLso_dNeGipqTUxEB4AaABAg   
492  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugy2jzgiLK8Ago5RJkZ4AaABAg   
493  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxMq_0ZWDifcQE7z4Z4AaABAg   
494  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugzia6tbZ_wyYVmKhBt4AaABAg   
495  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzlpBICovSgkLMggbJ4AaABAg   
496  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugy6IxZWqfz_6HCt2o94AaABAg   
497  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzNuMEHgheOHs6qDEp4AaABAg   
498  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugyz-PLnohHvnDh6VAp4AaABAg   
499  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwEDftvvE9Yu1x-hDd4AaABAg   

                author                                       text_comment  \
489    @sarahtania3456  Visi misi 02,03 hampir sama..mereka hebat bgt ...   
490   @yatisolyati2165  Asalamualaikum   pa  pravowo   mas  gibran  se...   
491        @joarva6480  terindikasi korupsi ektp.. tapi janji berantas...   
492      @user-4anyut4  <a href="https://www.youtube.com/watch?v=uNTbV...   
493  @MasKholidKebumen                       Saya dan keluarga pilih AMIN   
494         @Wafacilik  Semoga visi dan misinya nanti terlaksana bagi ...   
495    @sarahtania3456                      Kerja nyata bgt 03 masyaallah   
496     @dennytula5593  Saya pilih prabowo mungkin jika dia menang bis...   
497        @khoirul_a6                                           Hhhhhhhh   
498   @mochzakaria8521                 program nya gk ada yg masuk akal😮😢   
499       @fahimhd2141  banyak yg bilang anies banyak omong, yap betul...   

     like_count  total_reply                               channel_url  \
489           0            0    http://www.youtube.com/@sarahtania3456   
490           0            0   http://www.youtube.com/@yatisolyati2165   
491           0            0        http://www.youtube.com/@joarva6480   
492           0            0      http://www.youtube.com/@user-4anyut4   
493           0            0  http://www.youtube.com/@MasKholidKebumen   
494           0            0         http://www.youtube.com/@Wafacilik   
495           0            0    http://www.youtube.com/@sarahtania3456   
496           0            0     http://www.youtube.com/@dennytula5593   
497           0            0        http://www.youtube.com/@khoirul_a6   
498           0            0   http://www.youtube.com/@mochzakaria8521   
499           0            0       http://www.youtube.com/@fahimhd2141   

         published_date        updated_date  
489 2024-02-11 14:39:56 2024-02-11 14:39:56  
490 2024-02-11 14:42:14 2024-02-11 14:42:14  
491 2024-02-11 14:43:25 2024-02-11 14:43:25  
492 2024-02-11 14:56:33 2024-02-11 14:56:33  
493 2024-02-11 14:56:51 2024-02-11 14:56:51  
494 2024-02-11 14:59:02 2024-02-11 14:59:02  
495 2024-02-11 15:09:49 2024-02-11 15:09:49  
496 2024-02-11 15:26:17 2024-02-11 15:26:17  
497 2024-02-11 15:32:50 2024-02-11 15:32:50  
498 2024-02-11 15:51:09 2024-02-11 15:51:09  
499 2024-02-10 07:59:39 2024-02-10 07:59:39

## 2. Combining Five DataFrame Variables Contain Comments Information from the First - Fifth Debate Videos on KOMPASTV YouTube Channel

In [13]:
"""
Menggabungkan 5 variabel DataFrame menjadi 1 file dataset dalam format
spreadsheet Microsoft Excel dimana masing-masing variabel DataFrame memiliki jumlah
data yang disimpan berbeda-beda karena saat jumlah data yang didefinisikan saat
melakukan proses scraping/pengambilan informasi-informasi komentar utama dengan metode pengulangan (looping)
dibuat berbeda antara jumlah data komentar dari video debat pertama, debat kedua, dst
"""

df_comment_video_debate_kompastv = pd.concat([first_debate_video_df, second_debate_video_df, third_debate_video_df,
                                              fourth_debate_video_df, fifth_debate_video_df], ignore_index = True)

# Menampilkan 35 data teratas yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kompastv"
df_comment_video_debate_kompastv.head(35)

video_id                channel_id                  comment_id  \
0   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzTOJaoDFAy69VEmW54AaABAg   
1   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzmTSpOSPp8Aaid8tN4AaABAg   
2   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgytfyaQRBKOJD1MWmJ4AaABAg   
3   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMdksVHA_TUd0b8zZ4AaABAg   
4   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzKN65yuzOuwoV25xR4AaABAg   
5   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwkGCjnkn4qiL2pUq54AaABAg   
6   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwHT6Z026fOcLqsxT54AaABAg   
7   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwEn2xOZrliv0jvyhp4AaABAg   
8   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwpSjBjJZnKzyn5PmN4AaABAg   
9   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugz0wZUiNXc6ccaFu7J4AaABAg   
10  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyGHImkaLkXkDAMqcZ4AaABAg   
11  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyU-X3Lsh3ILboVJ3Z4AaABAg   
12  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwZ5CQXcfQ989V1nWR4AaABAg   
13  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxmaRidkVx6ZNhtXwN4AaABAg   
14  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwWEGrP4EKIKmkiOqN4AaABAg   
15  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxOFelnwKe7HLPP1aN4AaABAg   
16  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyaFu1tTObuurbV-ol4AaABAg   
17  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxaFx3JHpVIapD10S94AaABAg   
18  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugy7In-j-fEWYEmivrJ4AaABAg   
19  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugz44hx8zC79L1OTHWd4AaABAg   
20  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwjqAlc_Ei9wHRT0A94AaABAg   
21  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugxd9sibvHqg_kkjn_N4AaABAg   
22  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMPWdvd-urv1CjC1x4AaABAg   
23  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzkuDUgMUIQRv7Fvot4AaABAg   
24  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugwarfje623O694DOKJ4AaABAg   
25  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzrJP46lxWpF7lsYTd4AaABAg   
26  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyOmbKfPj-MB1FFNIZ4AaABAg   
27  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwqPUGLJwMnvpaNwux4AaABAg   
28  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugzx5I4JHk8B-jKdj_R4AaABAg   
29  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyG_xGiWFyatsEbzGF4AaABAg   
30  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugw-EWHpbIcERxmCQ8t4AaABAg   
31  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyxRzV3ElPIEj69ZRF4AaABAg   
32  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugx486knrOcdny8l-714AaABAg   
33  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxFBB4Aox4ZvVGdbfZ4AaABAg   
34  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgxwDROILzJOA2BFFxZ4AaABAg   

                       author  \
0          @alkhasanisani3619   
1             @MrGaming-fg9qu   
2          @desyanggraeni9380   
3           @aprianisalam7842   
4           @nanananina-eg5vo   
5        @RidhoRamadhan-gm6rp   
6            @user-mf1tc7co8s   
7           @nanasepdiana5531   
8                @FauzanFikli   
9             @titisuchan3357   
10           @PutriPeni-bx3rs   
11       @ahmanalsebayang6366   
12                @nana.edu23   
13      @tatytatydjulaeha2395   
14              @Rissachannel   
15       @JihadSaepuloh-sf9xt   
16           @niosubaktio1995   
17          @hadisuhendro8979   
18  @anisapurialdamayanti7148   
19        @SigitBahtiar-on6gs   
20             @hunterrevew94   
21           @susioktavia2369   
22      @WidhiannisaUlhusna96   
23    @dicky_dentajourney5477   
24          @hambaAllah-ph5gf   
25        @financecompany1582   
26                @prill_syaa   
27       @procrastinating3859   
28           @risnaholisa6882   
29               @susiuci5376   
30             @rudiwibowo975   
31          @SitiRohana-sq1zs   
32           @user-xg6rt4yj1r   
33            @ryanaudrey2653   
34       @muis_azwarazwar5171   

                                         text_comment  like_count  \
0   Anies pinter ngomong<br>Ganjar pinter jawab<br...         772   
1   Indonesia tidak kekurangan orang orang pintar&.

Berdasarkan code diatas, saya telah berhasil menggabungkan 5 variabel DataFrame yang masing-masing DataFrame memiliki data/informasi komentarnya masing-masing. Sebagai contoh pada variabel DataFrame "first_debate_video_df" berisikan 10.500 data/informasi komentar yang berhasil diambil dari video debat babak pertama capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel KOMPASTV. Sedangkan variabel DataFrame "second_debate_video_df" juga memiliki data/informasi komentar utama namun yang membedakannya dengan DataFrame "first_debate_video_df" adalah jumlah data yang diambil dan sumber videonya karena jika variabel DataFrame "second_debate_video_df" memiliki 500 dan sumber informasi/data komentar yang diambil berasal dari video debat babak kedua capres-cawapres Republik Indonesia tahun 2024 yang di-upload oleh YouTube channel KOMPASTV.

Hal tersebut juga sama dengan yang dimiliki oleh variabel DataFrame lainnya seperti "third_debate_video_debate_df", "fourth_debate_video_df" dan "fifth_debate_video_df" dengan catatan jumlah data yang diambil berbeda-beda dan sumber video atas informasi komentar yang diambil juga berbeda namun video-video tersebut tetap di-upload/ditayangkan oleh YouTube channel KOMPASTV.

Untuk mengecek dan memastikan apakah penggabungan 5 variabel DataFrame berhasil dilakukan, saya akan menampilkan 11 data terbawah yang dimiliki oleh DataFrame "df_comment_video_debate_kompastv" karena jika proses penggabungan 5 variabel DataFrame menjadi 1 DataFrame berhasil dilakukan maka 11 data terbawah seharusnya berasal dari data-data komentar yang diambil (di-scraping) dari video debat babak kelima capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel KOMPASTV dan disimpan di dalam variabel DataFrame bernama "fifth_debate_video_df".

Sebagai penjelasan tambahan saat saya melakukan proses penggabungan 5 variabel DataFrame yang berbeda-beda menjadi 1 DataFrame agar data-data komentar dari video debat babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KOMPASTV tidak terpisah-pisah dan tersentralisasi (terpusat) ke dalam 1 file dataset, saya menggunakan sebuah parameter yaitu ignore_index yang bernilai Boolean yaitu True. Parameter tersebut saya gunakan (ignore_index = True) bertujuan agar indeks pada DataFrame (dalam hal ini indeks yang terdapat di dalam variabel DataFrame "df_comment_video_debate_kompastv") tidak terduplikasi dan tidak mengulangi nilai indeks yang sama dari indeks awal.

Sebagai contoh variabel DataFrame "first_video_debate_df" memiliki sebuah nilai awal indeks yaitu 0 (indeks data 0 mengartikan jika data tersebut merupakan baris pertama yang ada di dalam data tersebut) dan indeks terakhir pada variabel DataFrame "first_video_debate_df" adalah 10499 yang mengartikan bahwa baris data tersebut merupakan baris data ke-10.500. Lalu karena DataFrame "first_video_debate_df" digabungkan dengan DataFrame "second_video_debate_df" maka baris data selanjutnya sudah pasti merupakan data yang dimiliki oleh indeks data ke-0/baris pertama dari DataFrame "second_video_debate_df".

Jika tidak menggunakan parameter ignore_index yang bernilai True, maka variabel DataFrame "df_comment_video_debate_kompastv" yang telah menjadi DataFrame tersentralisasi tidak melanjutkan nilai indeks yang selanjutnya yaitu indeks data ke-10500 yang merupakan data pertama/indeks data ke-0 dari variabel DataFrame "second_video_debate_df". Jadi secara tidak langsung akan ada 2 baris data yang memiliki indeks ke-0 jika DataFrame "df_comment_video_debate_kompastv" berisikan data-data komentar dari DataFrame "first_video_debate_df" dan "second_video_debate_df". Karena DataFrame yang tersentralisasi ini berisikan data-data komentar hasil penggabungan 5 variabel DataFrame yang berbeda, maka jika tidak menggunakan parameter ignore_index bernilai True DataFrame "df_comment_video_debate_kompastv" memiliki 5 baris data yang memiliki indeks data ke-0

Jadi alasan tersebutlah saya menggunakan parameter ignore_index = True untuk menghindari terjadinya duplikasi indeks data pada DataFrame yang tersentralisasi/terpusat yaitu variabel DataFrame "df_comment_video_debate_kompastv".

In [14]:
# Menampilkan 11 data terbawah yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kompastv"
df_comment_video_debate_kompastv.tail(11)

video_id                channel_id                  comment_id  \
12489  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugxb0kUlJ-q5lKD5Urd4AaABAg   
12490  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxTJgAR-MZOsCb_GvR4AaABAg   
12491  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzLso_dNeGipqTUxEB4AaABAg   
12492  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugy2jzgiLK8Ago5RJkZ4AaABAg   
12493  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgxMq_0ZWDifcQE7z4Z4AaABAg   
12494  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugzia6tbZ_wyYVmKhBt4AaABAg   
12495  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzlpBICovSgkLMggbJ4AaABAg   
12496  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugy6IxZWqfz_6HCt2o94AaABAg   
12497  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgzNuMEHgheOHs6qDEp4AaABAg   
12498  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  Ugyz-PLnohHvnDh6VAp4AaABAg   
12499  uNTbVZ-YyQo  UC5BMIWZe9isJXLZZWPWvBlg  UgwEDftvvE9Yu1x-hDd4AaABAg   

                  author                                       text_comment  \
12489    @sarahtania3456  Visi misi 02,03 hampir sama..mereka hebat bgt ...   
12490   @yatisolyati2165  Asalamualaikum   pa  pravowo   mas  gibran  se...   
12491        @joarva6480  terindikasi korupsi ektp.. tapi janji berantas...   
12492      @user-4anyut4  <a href="https://www.youtube.com/watch?v=uNTbV...   
12493  @MasKholidKebumen                       Saya dan keluarga pilih AMIN   
12494         @Wafacilik  Semoga visi dan misinya nanti terlaksana bagi ...   
12495    @sarahtania3456                      Kerja nyata bgt 03 masyaallah   
12496     @dennytula5593  Saya pilih prabowo mungkin jika dia menang bis...   
12497        @khoirul_a6                                           Hhhhhhhh   
12498   @mochzakaria8521                 program nya gk ada yg masuk akal😮😢   
12499       @fahimhd2141  banyak yg bilang anies banyak omong, yap betul...   

       like_count  total_reply                               channel_url  \
12489           0            0    http://www.youtube.com/@sarahtania3456   
12490           0            0   http://www.youtube.com/@yatisolyati2165   
12491           0            0        http://www.youtube.com/@joarva6480   
12492           0            0      http://www.youtube.com/@user-4anyut4   
12493           0            0  http://www.youtube.com/@MasKholidKebumen   
12494           0            0         http://www.youtube.com/@Wafacilik   
12495           0            0    http://www.youtube.com/@sarahtania3456   
12496           0            0     http://www.youtube.com/@dennytula5593   
12497           0            0        http://www.youtube.com/@khoirul_a6   
12498           0            0   http://www.youtube.com/@mochzakaria8521   
12499           0            0       http://www.youtube.com/@fahimhd2141   

           published_date        updated_date  
12489 2024-02-11 14:39:56 2024-02-11 14:39:56  
12490 2024-02-11 14:42:14 2024-02-11 14:42:14  
12491 2024-02-11 14:43:25 2024-02-11 14:43:25  
12492 2024-02-11 14:56:33 2024-02-11 14:56:33  
12493 2024-02-11 14:56:51 2024-02-11 14:56:51  
12494 2024-02-11 14:59:02 2024-02-11 14:59:02  
12495 2024-02-11 15:09:49 2024-02-11 15:09:49  
12496 2024-02-11 15:26:17 2024-02-11 15:26:17  
12497 2024-02-11 15:32:50 2024-02-11 15:32:50  
12498 2024-02-11 15:51:09 2024-02-11 15:51:09  
12499 2024-02-10 07:59:39 2024-02-10 07:59:39

Berdasarkan hasil diatas dapat diketahui jika indeks data ke-12299 atau yang merupakan baris data 12.299 sama dengan data komentar yang dimiliki oleh variabel DataFrame "fifth_video_debate_df" yang merupakan indeks data ke-499 atau pada baris data 500 sehingga proses penggabungan data-data komentar yang didapatkan dari video debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia tahun 2024 yang setiap video debat ditayangkan oleh channel YouTube KOMPASTV telah berhasil dilakukan dan urutan indeks dari variabel DataFrame yang tersentralisasi (df_comment_video_debate_kompastv) sudah sesuai dengan output yang diharapkan.

In [15]:
# Menampilkan jumlah data duplikat yang dimiliki di dalam variabel DataFrame "df_comment_video_debate_kompastv"
df_comment_video_debate_kompastv.duplicated().sum()

0

In [16]:
"""
Menampilkan jumlah nilai data yang hilang (missing values) dari masing-masing
kolom yang terdapat di dalam variabel DataFrame "df_comment_video_debate_kompastv"
"""
df_comment_video_debate_kompastv.isnull().sum()

video_id          0
channel_id        0
comment_id        0
author            0
text_comment      0
like_count        0
total_reply       0
channel_url       0
published_date    0
updated_date      0
dtype: int64

In [17]:
"""
Menampilkan beberapa informasi dasar dari setiap kolom seperti tipe data,
jumlah baris data atau kolom yang dimiliki oleh DataFrame, dll., yang dimiliki
oleh variabel DataFrame "df_comment_video_debate_kompastv"
"""
df_comment_video_debate_kompastv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12500 entries, 0 to 12499
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        12500 non-null  object        
 1   channel_id      12500 non-null  object        
 2   comment_id      12500 non-null  object        
 3   author          12500 non-null  object        
 4   text_comment    12500 non-null  object        
 5   like_count      12500 non-null  int64         
 6   total_reply     12500 non-null  int64         
 7   channel_url     12500 non-null  object        
 8   published_date  12500 non-null  datetime64[ns]
 9   updated_date    12500 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 976.7+ KB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "df_comment_video_debate_kompastv" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "df_comment_video_debate_kompastv" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolom dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden-Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KOMPASTV.

2. Variabel DataFrame "df_comment_video_debate_kompastv" memiliki jumlah data/baris sebanyak 12.500 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang masing-masing memiliki jumlah data/informasi komentar yang berbeda-beda seperti pada variabel DataFrame "first_debate_video_df" memiliki baris data sebanyak 10.500 informasi komentar, variabel DataFrame "second_debate_video_df" memiliki baris data sebanyak 500 data komentar, dst.

3. DataFrame "df_comment_video_debate_kompastv" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "df_comment_video_debate_kompastv" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kompastv" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "df_comment_video_debate_kompastv" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "df_comment_video_debate_kompastv" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kompastv" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "df_comment_video_debate_kompastv" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

In [18]:
"""
Mengubah urutan data yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kompastv"
berdasarkan jumlah like terbanyak hingga jumlah like paling sedikit yang didapatkan/diterima oleh suatu teks komentar
diantara 12.500 data komentar yang merupakan kumpulan data hasil penggabungan komentar-komentar dari video
debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia yang ditayangkan oleh channel YouTube KOMPASTV
"""

df_comment_video_debate_kompastv = df_comment_video_debate_kompastv.sort_values(by=["like_count"],
                                                                                ascending = False, ignore_index = True)

df_comment_video_debate_kompastv.head(13) # Menampilkan 13 data komentar teratas yang memiliki jumlah like terbanyak

video_id                channel_id                  comment_id  \
0   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzTOJaoDFAy69VEmW54AaABAg   
1   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzmTSpOSPp8Aaid8tN4AaABAg   
2   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgytfyaQRBKOJD1MWmJ4AaABAg   
3   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyMdksVHA_TUd0b8zZ4AaABAg   
4   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgzKN65yuzOuwoV25xR4AaABAg   
5   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwkGCjnkn4qiL2pUq54AaABAg   
6   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwHT6Z026fOcLqsxT54AaABAg   
7   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwEn2xOZrliv0jvyhp4AaABAg   
8   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwpSjBjJZnKzyn5PmN4AaABAg   
9   zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  Ugz0wZUiNXc6ccaFu7J4AaABAg   
10  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyGHImkaLkXkDAMqcZ4AaABAg   
11  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgyU-X3Lsh3ILboVJ3Z4AaABAg   
12  zB0RvKAc2kw  UC5BMIWZe9isJXLZZWPWvBlg  UgwZ5CQXcfQ989V1nWR4AaABAg   

                  author                                       text_comment  \
0     @alkhasanisani3619  Anies pinter ngomong<br>Ganjar pinter jawab<br...   
1        @MrGaming-fg9qu  Indonesia tidak kekurangan orang orang pintar&...   
2     @desyanggraeni9380  Disaat Ganjar dan Anies saling menyerang Prabo...   
3      @aprianisalam7842  Pak Prabowo memang kurang dr hal seperti pak g...   
4      @nanananina-eg5vo  Jendral tidak dilatih untuk merangkai kata2 ma...   
5   @RidhoRamadhan-gm6rp  Saya tau rasanya menjadi orang seperti pak pro...   
6       @user-mf1tc7co8s  Pak prabowo tu keras , disiplin karena mantan ...   
7      @nanasepdiana5531  Pak prabowo tidak pandai merangkai kata - kata...   
8           @FauzanFikli  Saya yakin .pak Prabowo sudah tahu keadaan neg...   
9        @titisuchan3357  P Prabowo tidak menjatuhkan lawan,,tetap legow...   
10      @PutriPeni-bx3rs  pak prabowo tidak pandai merangkai kata,tapi d...   
11  @ahmanalsebayang6366  Anis pintar memgelola kata<br>Cocok sekali jad...   
12           @nana.edu23  Pak Prabowo retorika nya sederhana, namun muda...   

    like_count  total_reply                                  channel_url  \
0          772          154    http://www.youtube.com/@alkhasanisani3619   
1          607           43       http://www.youtube.com/@MrGaming-fg9qu   
2          474           86    http://www.youtube.com/@desyanggraeni9380   
3          450           77     http://www.youtube.com/@aprianisalam7842   
4          431           56     http://www.youtube.com/@nanananina-eg5vo   
5          409           28  http://www.youtube.com/@RidhoRamadhan-gm6rp   
6          338           46      http://www.youtube.com/@user-mf1tc7co8s   
7          330           38     http://www.youtube.com/@nanasepdiana5531   
8          318           36          http://www.youtube.com/@FauzanFikli   
9          312           51       http://www.youtube.com/@titisuchan3357   
10         293           43      http://www.youtube.com/@PutriPeni-bx3rs   
11         288           61  http://www.youtube.com/@ahmanalsebayang6366   
12         284           12           http://www.youtube.com/@nana.edu23   

        published_date        updated_date  
0  2023-12-13 13:26:17 2023-12-13 13:26:17  
1  2023-12-23 15:19:21 2023-12-23 15:19:21  
2  2023-12-15 09:21:02 2023-12-15 09:21:36  
3  2023-12-14 12:31:52 2023-12-14 12:31:52  
4  2023-12-17 03:43:34 2023-12-17 03:43:34  
5  2023-12-27 14:30:09 2023-12-27 14:30:09  
6  2023-12-14 01:33:20 2023-12-14 01:33:20  
7  2023-12-26 08:02:26 2023-12-26 08:02:26  
8  2023-12-15 16:14:53 2023-12-15 16:14:53  
9  2023-12-15 09:35:11 2023-12-15 09:35:11  
10 2023-12-23 05:38:09 2023-12-23 05:38:09  
11 2023-12-13 17:21:14 2023-12-13 17:21:14  
12 2023-12-15 12:01:44 2023-12-15 12:01:44

## 3. Export DataFrame Variable into Dataset with Spreadsheet Format (Microsoft Excel)

In [19]:
"""
Melakukan proses penyimpanan/meng-eksport data atau informasi komentar utama sebanyak 12.500 dari hasil
penggabungan data-data dari kelima variabel DataFrame yang didapatkan dari hasil web scraping pada
video debat capres-cawapres babak pertama hingga babak kelima yang di-upload oleh Channel YouTube KOMPASTV
"""

df_comment_video_debate_kompastv.to_excel("../Dataset/Dataset Komentar Video Debat dari YouTube Channel KOMPASTV.xlsx",
                                          index = False, header = True)